tests/integration/ha/test_storage.py

```diff
@@ -172,3 +172,81 @@ async def test_storage_reuse_after_scale_down(
     # check if data is also imported
     assert writes_result.count == (await c_writes.count())
     assert writes_result.max_stored_id == (await c_writes.max_stored_id())
+
+
+@pytest.mark.abort_on_fail
+async def test_storage_reuse_in_new_cluster_after_app_removal(
+    ops_test: OpsTest, c_writes: ContinuousWrites, c_balanced_writes_runner
+):
+    """Check storage is reused and data accessible after removing app and deploying new cluster."""
+    app = (await app_name(ops_test)) or APP_NAME
+
+    if storage_type(ops_test, app) == "rootfs":
+        pytest.skip(
+            "re-use of storage can only be used on deployments with persistent storage not on rootfs deployments"
+        )
+
+    # scale-down to 1 if multiple units
```

Comment:
You use this code a few times. Perhaps it could be extracted to a function?

Reply:
thank you, will have an integration tests cleanup a bit later